cpp/libcudf_kafka/CMakeLists.txt

```diff
@@ -13,112 +13,76 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #=============================================================================
-cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
-
-project(CUDA_KAFKA VERSION 0.15.0 LANGUAGES C CXX CUDA)
-
-# TODO: Since we have no actual CUDA code in cudf_kafka this should be removed in the future
-# in favor of using FindCUDAToolkit to get the needed CUDA include headers
-if(NOT CMAKE_CUDA_COMPILER)
-  message(SEND_ERROR "CMake cannot locate a CUDA compiler")
-endif(NOT CMAKE_CUDA_COMPILER)
+cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
+project(CUDA_KAFKA VERSION 0.19.0 LANGUAGES C CXX)
+
+#################################################################################################
+# - CUDA architectures
+
+# If CMAKE_CUDA_ARCHITECTURES is not defined, build for all supported architectures. If
+# CMAKE_CUDA_ARCHITECTURES is set to an empty string (""), build for only the current
+# architecture. If `CMAKE_CUDA_ARCHITECTURES` is specified by the user, use user setting.
+
+# This needs to be run before enabling the CUDA language due to the default initialization behavior
+# of CMAKE_CUDA_ARCHITECTURES, https://gitlab.kitware.com/cmake/cmake/-/issues/21302
+if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+  set(CUDF_BUILD_FOR_ALL_ARCHS TRUE)
+elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "")
+  unset(CMAKE_CUDA_ARCHITECTURES CACHE)
```

Comment:
This line ( 31 )  isn't needed anymore since you are using `cudf/cpp` GPU eval code.